include/dd/Package.hpp

```diff
@@ -595,6 +668,75 @@ namespace dd {
         }
 
+        std::tuple<CMat, CMat, CMat, CMat> quarterMatrix(const CMat& matrix) {
+            /**
+
+            Splits a given square matrix into four smaller matrices, representing the four quadrants of the original matrix.
+            @param matrix the matrix to be split into quadrants.
+            @return a tuple of four matrices representing the four quadrants of the original matrix.
+            */
+            const auto length = matrix.size();
+            const auto half   = length / 2;
+
+            CMat quad0 = CMat(static_cast<size_t>(half), CVec(static_cast<size_t>(half), {0.0, 0.0}));
+            CMat quad1 = CMat(static_cast<size_t>(half), CVec(static_cast<size_t>(half), {0.0, 0.0}));
+            CMat quad2 = CMat(static_cast<size_t>(half), CVec(static_cast<size_t>(half), {0.0, 0.0}));
+            CMat quad3 = CMat(static_cast<size_t>(half), CVec(static_cast<size_t>(half), {0.0, 0.0}));
+
+            for (std::vector<double>::size_type i = 0; i < length; ++i) {
+                for (std::vector<double>::size_type j = 0; j < length; ++j) {
+                    if (i < half && j < half) {
+                        quad0[i % half][j % half] = matrix[i][j];
+                    } else if (i < half && j >= half) {
+                        quad1[i % half][j % half] = matrix[i][j];
+                    } else if (i >= half && j < half) {
+                        quad2[i % half][j % half] = matrix[i][j];
+                    } else if (i >= half && j >= half) {
+                        quad3[i % half][j % half] = matrix[i][j];
+                    }
+                }
+            }
+
+            return std::tuple{quad0, quad1, quad2, quad3};
+        }
+
+        mEdge makeDDFromMatrix(const CVec::const_iterator& begin,
+                               const CVec::const_iterator& end,
+                               const Qubit                 level) {
+            /**
+
+            Converts a vectorized matrix into a decision diagram starting from the top qubit level.
+             @param begin an iterator to the beginning of the range of complex numbers representing the matrix.
+             @param end an iterator to the end of the range of complex numbers representing the matrix.
+             @param level the qubit level at which to start building the DD.
+             @return a DD representing the matrix.
+            */
+            if (level == 0) {
+                const auto& zeroWeight  = cn.getCached(begin->real(), begin->imag());
+                auto        element     = std::next(begin);
+                const auto& oneWeight   = cn.getCached(element->real(), element->imag());
+                element                 = std::next(element);
+                const auto& twoWeight   = cn.getCached(element->real(), element->imag());
+                element                 = std::next(element);
+                const auto& threeWeight = cn.getCached(element->real(), element->imag());
```

Comment:
Also just a side note for the future:
```suggestion
                auto elementIt = begin;
                const auto& zeroWeight  = cn.getCached(*elementIt);
                ++elementIt;
                const auto& oneWeight   = cn.getCached(*elementIt);
                ++elementIt;
                const auto& twoWeight   = cn.getCached(*elementIt);
                ++elementIt;
                const auto& threeWeight = cn.getCached(*elementIt);
```
would be the cleaner C++ way. (given the new `cn.getCached` overload)

Reply:
Function removed, but noted for the future